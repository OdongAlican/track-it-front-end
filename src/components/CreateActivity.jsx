import React, { useState } from 'react'
import { createActivity } from '../actions/activitiesAction'
import { useDispatch } from 'react-redux'
import Footer from '../components/Footer'

const CreateActivity = ({ history }) => {
    const [title, setTitle] = useState('')
    const [total, setTotal] = useState('')
    const [image, setImage] = useState(null)
    const dispatch = useDispatch()

    const submitAcvtivty = (event) => {
        
        const form = new FormData()
        form.append("title", title)
        form.append("total", total)
        form.append("avatar", image)
        dispatch(createActivity(form, history))
        event.preventDefault()

    }

    return (
        <div>
        <div className="card col-md-3 p-3">
            <h3>Create Activity</h3>
            <form onSubmit={ submitAcvtivty }>
                <input 
                type="text"
                placeholder="Enter Title"
                value = { title } 
                onChange={ e => setTitle(e.target.value) } 
                className="form-control mb-2"></input>
                <input 
                type="number"
                placeholder="Enter Total"
                value = { total } 
                onChange={ e => setTotal(e.target.value) } 
                className="form-control"></input>
                <input
                type='file'
                className="mt-2"
                onChange={(e) => setImage(e.target.files[0])}
                ></input>
                <button type="submit" className="btn btn-primary mt-1">
                    Create
                </button>
            </form>
        </div>
        <Footer/>
        </div>
    )
}

export default CreateActivity