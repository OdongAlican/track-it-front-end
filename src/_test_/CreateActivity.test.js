import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import CreateActivity from '../components/CreateActivity';

afterEach(cleanup);
const initialState = {
  authReducer: {
    output: 10,
    error: 'test',
  },
};

const mockStore = configureStore();
let store;
const props = {
  history: '/',
};

it('renders CreatActivity correctly', () => {
  store = mockStore(initialState);
  const { getAllByTestId } = render(
    <Router>
      <Provider store={store}><CreateActivity props={props} /></Provider>
      ,
    </Router>,
  );
  const result = getAllByTestId('appCreateActivity');
  expect(result[0])
    .toHaveTextContent('You can add a photo to your activityCreateHomeCreate ActivityInformationMore');
});

it('renders CreatActivity correctly', () => {
  store = mockStore(initialState);
  const { getAllByTestId } = render(
    <Router>
      <Provider store={store}><CreateActivity props={props} /></Provider>
      ,
    </Router>,
  );
  const result = getAllByTestId('appCreateActivity');
  expect(result[0]).not
    .toHaveTextContent('Some Random Text');
});

it('matches CreatActivity snapshot', () => {
  const appointment = renderer.create(
    <Router>
      <Provider store={store}><CreateActivity props={props} /></Provider>
      ,
    </Router>,
  );
  const tree = appointment.toJSON();
  expect(tree).toMatchSnapshot();
});
