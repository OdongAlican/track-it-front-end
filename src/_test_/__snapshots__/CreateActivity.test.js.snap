// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches CreatActivity snapshot 1`] = `
Array [
  <div
    className="create-activity"
    data-testid="appCreateActivity"
  >
    <div
      className="create-activity-inner"
    >
      <div
        className="activity-heading mb-2 p-3"
      >
        <span>
          You can add a photo to your activity
        </span>
      </div>
      <div
        className="image-upload m-3 p-2 d-flex"
      >
        <div
          className="left-image"
        >
          <i
            className="fas fa-tv"
          />
        </div>
        <div
          className="right-image"
        >
          <i
            className="fas fa-church"
          />
        </div>
      </div>
      <div>
        <form
          onSubmit={[Function]}
        >
          <input
            className="input-control mb-2"
            onChange={[Function]}
            placeholder="Enter Title"
            type="text"
            value=""
          />
          <input
            className="input-control"
            onChange={[Function]}
            placeholder="Enter Total"
            type="number"
            value=""
          />
          <input
            className="mt-2 btn-control file-section"
            onChange={[Function]}
            type="file"
          />
          <button
            className="btn-control mt-1"
            type="submit"
          >
            Create
          </button>
        </form>
      </div>
    </div>
    <div
      className="footer"
      data-testid="appFooter"
    >
      <div
        className="link-div px-0"
        data-testid="appButton"
      >
        <button
          className="link-button"
          type="button"
        >
          <a
            href="/activities"
            onClick={[Function]}
            style={
              Object {
                "color": "#ABAEB3",
              }
            }
          >
            <i
              className="far fa-chart-bar footer-btn"
            />
            <p>
              Home
            </p>
          </a>
        </button>
      </div>
      <div
        className="link-div px-0"
        data-testid="appButton"
      >
        <button
          className="link-button"
          type="button"
        >
          <a
            href="/create-activity"
            onClick={[Function]}
            style={
              Object {
                "color": "#ABAEB3",
              }
            }
          >
            <i
              className="fas fa-chart-line footer-btn"
            />
            <p>
              Create Activity
            </p>
          </a>
        </button>
      </div>
      <div
        className="link-div px-0"
        data-testid="appButton"
      >
        <button
          className="link-button"
          type="button"
        >
          <a
            href="/information"
            onClick={[Function]}
            style={
              Object {
                "color": "#ABAEB3",
              }
            }
          >
            <i
              className="fas fa-chart-pie footer-btn"
            />
            <p>
              Information
            </p>
          </a>
        </button>
      </div>
      <div
        className="link-div px-0"
        data-testid="appButton"
      >
        <button
          className="link-button"
          type="button"
        >
          <a
            href="/activities"
            onClick={[Function]}
            style={
              Object {
                "color": "#ABAEB3",
              }
            }
          >
            <i
              className="fas fa-ellipsis-h footer-btn"
            />
            <p>
              More
            </p>
          </a>
        </button>
      </div>
    </div>
  </div>,
  ",",
]
`;
